src/Dropdown/Dropdown.svelte

```diff
@@ -245,6 +237,7 @@
           open = false;
         }
       }}"
+      on:blur="{() => (open = false)}"
```

Comment:
The flaw with adding a blur handler is it will run before a user can select an option. Instead, we can add an event listener to the parent window during `onMount`:

```html
<script>

  // ...

  const pageClickHandler = ({ target }) => {
    if (open && ref && !ref.contains(target)) {
      open = false;
    }
  };

  onMount(() => {
    if (parent) {
      parent.addEventListener("click", pageClickHandler);
    }
  });
</script>

<svelte:window on:click="{pageClickHandler}" />
```

Reply:
I don't think we need to avoid blur completly. You proposition will work if there's only one `iframe` but will fail with mutpliple. To make the code works, we can use the `mousedown` event instead of the `click` [6b84b51](https://github.com/carbon-design-system/carbon-components-svelte/pull/1596/commits/6b84b51238ad3afcb96926dfad925b3ca87d9d3b)

If you want to use `parent` I can make it works with a loop for parents.

Reply:
`on:mousedown` is an interesting solution as it works on both touch and pointer devices. However, this will cause Dropdown items to become selected when the user clicks down instead of `mouseup`, which may be unexpected behaviour to the user.

It seems `on:blur` works against us. The only alternative I see is listening for `on:blur`, but also listen for `mousedown` and set a flag if the user is currently clicking on a list item, though I fear a race condition may occur. I got something that works:

```html
<script>
  // ...

  let active = false;
</script>

<button
  on:blur="{() => {
    if (active) return;
    open = false;
  }}"
>
<!-- -->
</button>

<!-- -->
<ListBoxMenuItem
  on:mousedown="{() => {
    active = true;
  }}"
  on:click="{(e) => {
    // ... assign values
    active = false;
  }}"
/>
```

It keeps the Dropdown 'active' when there's a mousedown. Try that out and let me know what you think. Any thoughts @metonym?

Note, changing code in ListBoxMenuItem will impact consumers MultiSelect and Combobox; but we can work on those once we agree on an interaction data flow.

Reply:
I made some tests and `onblur` won't work because of the `label` which would cause more problems. I think we'll need to stick with `svelte:window` origin. I'll post a commit with your  `pageClickHandler`.

Reply:
Sorry for all the commits. I'm still getting used to `git`. Someday I'll understand everything.

Reply:
What sort of issues wth `label` did you run into? Also don't worry about making so many commits, it's often good practice. 😃

Reply:
The `blur` event is set to the button so if the `label` is clicked, `blur` will be called and will flash the `ListItemMenu`.

Reply:
@theetrain I added `onDestroy` event to remove the the `eventListener` in case of the component get destroyed. We don't want to overpopulate the `eventListeners` :^)